modules/tinymce/Gruntfile.js

```diff
@@ -31,10 +31,18 @@ const stripSourceMaps = function (data) {
 };
 
 module.exports = function (grunt) {
-  var packageData = grunt.file.readJSON('package.json');
-  var changelogLine = grunt.file.read('changelog.txt').toString().split('\n')[0];
-  var BUILD_VERSION = packageData.version + (process.env.BUILD_NUMBER ? '-' + process.env.BUILD_NUMBER : '');
-  packageData.date = /^Version [^\(]+\(([^\)]+)\)/.exec(changelogLine)[1];
+  const packageData = grunt.file.readJSON('package.json');
+  const BUILD_VERSION = packageData.version + (process.env.BUILD_NUMBER ? '-' + process.env.BUILD_NUMBER : '');
+
+  // Determine the release date
+  const dateRe = new RegExp('^##\\s+' + packageData.version.toString().replace(/\./g, '\\.') + '\\s+\\-\\s+([\\d-]+)$', 'm');
```

Comment:
Do we ever actually need this regex to tell the difference between the first instance of `## [any old version string] - date` and `## [my specific version string] - date` in the changelog? From the old code it looks like we could get away with just going for the first line that has _a version_ on it.

If not, we could simplify the `packageData.version.toString().replace(/\./g, '\\.')` down to `[^\\s]+` and make the whole thing a bit simpler.

Reply:
We 100% need this to be the version matching package.json. With the new format, we aren't guaranteed that it'll be first, as it could pick up `Unversioned` potentially as well. So no, we definitely don't want to be looking at making this less strict.

Reply:
Unversioned wouldn't have a date next to it, but if there's general risks of that going wrong then lets leave it as is

Reply:
You can put a date beside it though, there's nothing stopping someone accidentally doing that. It wouldn't be correct, as unversioned shouldn't have a date, but having it pick that up instead would be quite bad.